libazureinit/src/provision/ssh.rs

```diff
@@ -67,20 +76,65 @@ pub(crate) fn get_authorized_keys_path_from_sshd() -> io::Result<Option<PathBuf>
     Ok(None)
 }
 
+pub(crate) fn update_sshd_config(sshd_config_path: &str) -> Result<(), io::Error> {
+    // Check if the path exists otherwise create it
+    let sshd_config_path = PathBuf::from(sshd_config_path);
+    if !sshd_config_path.exists() {
+        let mut file = std::fs::File::create(&sshd_config_path)?;
+        file.set_permissions(Permissions::from_mode(0o644))?;
+        file.write_all(b"PasswordAuthentication yes\n")?;
+        file.flush()?;
```

Comment:
I am curious why we call `flush()` after writing a file, because `File::flush()` for Linux is actually [no-op](https://github.com/rust-lang/rust/blob/27e38f8fc7efc57b75e9a763d7a0ee44822cd5f7/library/std/src/sys/pal/unix/fs.rs#L1310).
If we really want to ensure files to be written to underlying devices, `File::sync_all` or `File::sync_data` could be actually valid, but that would result in performance impact.

Reply:
I think we can avoid using flush and sync as well.